Error deleting 'project'

Create a project, give the project to users, then click the trash. The project is no longer displayed, but an error message is displayed.

![Screenshot](https://user-images.githubusercontent.com/3040657/82320009-d2615f00-99d2-11ea-8f92-8393659eb5b4.jpg)
Stacktrace:
```
[ERROR] 2020-05-19 08:41:23.782 [ajp-nio-8009-exec-3] SqlExceptionHelper - Cannot delete
    or update a parent row: a foreign key constraint fails (`kitodo`.`project_x_template`,
    CONSTRAINT `FK_project_x_template_project_id` FOREIGN KEY (`project_id`) REFERENCES
    `project` (`id`))
[DEBUG] 2020-05-19 08:41:23.845 [ajp-nio-8009-exec-3] SearchService -
    org.kitodo.data.database.exceptions.DAOException:
    javax.persistence.PersistenceException:
    org.hibernate.exception.ConstraintViolationException: could not execute statement
[ERROR] 2020-05-19 08:41:23.845 [ajp-nio-8009-exec-3] ProjectForm - errorDeleting
    org.kitodo.data.exceptions.DataException: 
    org.kitodo.data.database.exceptions.DAOException: 
    javax.persistence.PersistenceException: 
    org.hibernate.exception.ConstraintViolationException: could not execute statement
    at org.kitodo.production.services.data.base.SearchService.remove
        (SearchService.java:334)
    at org.kitodo.production.services.data.ProjectService.delete(ProjectService.java:377)
    at org.kitodo.production.forms.ProjectForm.delete(ProjectForm.java:291)
    ...
```
After logout and login, the index must be updated because projects are not fully indexed. It looks like the project was deleted from the index, but not from the database. However, it is no longer displayed. It is probably no longer possible to create a new project with the name.

**Goal:** Delete project completely. If this is not possible, do not delete the project at all.

--- a/Kitodo/src/main/java/org/kitodo/production/forms/ProjectForm.java
+++ b/Kitodo/src/main/java/org/kitodo/production/forms/ProjectForm.java
@@ -228,8 +228,6 @@ public class ProjectForm extends BaseForm {
         ServiceManager.getProjectService().evict(this.project);
         if (isTitleValid()) {
             try {
-                // call this to make saving and deleting permanent
-                this.commitFolders();
                 addFirstUserToNewProject();
 
                 ServiceManager.getProjectService().saveToDatabase(this.project);
@@ -250,6 +248,9 @@ public class ProjectForm extends BaseForm {
                 this.deletedTemples = new ArrayList<>();
 
                 ServiceManager.getProjectService().save(this.project, true);
+
+                // call this to make saving and deleting permanent
+                this.commitFolders();
 
                 return projectsPage;
             } catch (DAOException | DataException e) {

